### PR TITLE
[shared_preferences] Endorse shared_preferences

### PR DESCRIPTION
**Pull request by franciscojma86**
## Description

Endorses shared_preferences_macos by adding it as a platform in pubspec.yaml.
Also adds a macos folder to the example to support to support flutter run
Adds dummy macos podspec as a workaround for flutter/flutter#46618

## Related Issues
https://github.com/flutter/flutter/issues/41718

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

### CODE REVIEW

File: packages/shared_preferences/shared_preferences/pubspec.yaml [hunk 1]
```diff
@@ -2,7 +2,7 @@ name: shared_preferences
 description: Flutter plugin for reading and writing simple key-value pairs.
   Wraps NSUserDefaults on iOS and SharedPreferences on Android.
 homepage: https://github.com/flutter/plugins/tree/master/packages/shared_preferences/shared_preferences
-version: 0.5.4+10
+version: 0.6.0
```
Comment (amirh) [inline packages/shared_preferences/shared_preferences/pubspec.yaml]: Should be a minor version update

File: packages/shared_preferences/shared_preferences/pubspec.yaml [hunk 2]
```diff
@@ -12,12 +12,21 @@ flutter:
         pluginClass: SharedPreferencesPlugin
       ios:
         pluginClass: FLTSharedPreferencesPlugin
+      macos:
+        default_package: shared_preferences_macos
+        pluginClass: SharedPreferencesPlugin
```
Comment (amirh) [inline packages/shared_preferences/shared_preferences/pubspec.yaml]: I believe we shouldn't have pluginClass here

Comment (franciscojma86) [reply packages/shared_preferences/shared_preferences/pubspec.yaml]: Done

File: packages/shared_preferences/shared_preferences/pubspec.yaml (hunk 1 above)
Comment (franciscojma86) [reply packages/shared_preferences/shared_preferences/pubspec.yaml]: Done

**Pull request merged**